smart-contracts/contracts/cl-vault/src/vault/deposit.rs

```diff
@@ -247,13 +249,13 @@ fn refund_bank_msg(
     denom0: String,
     denom1: String,
 ) -> Result<Option<(BankMsg, Vec<Attribute>)>, ContractError> {
+    // Notice: checked_sub has been replaced with saturating_sub due to overflowing response from dex
     let refund0 = current_deposit
         .token0_in
-        .checked_sub(Uint128::new(resp.amount0.parse::<u128>()?))?;
-
+        .saturating_sub(Uint128::new(resp.amount0.parse::<u128>()?));
```

Comment:
Same for this saturating_sub

Reply:
[`83cbd87`](https://github.com/quasar-finance/quasar/pull/537/commits/83cbd87458b0beea209d65fdce91cf0511a712cf)